src/components/node-list/node-list-items.js

```diff
@@ -330,3 +349,314 @@ export const getFilteredItems = createSelector(
+ * @param {obj} modularPipeline the modular pipeine that needs the construction of a modular pipeline item
+ * @return {obj} modular pipeline item
+ */
+const constructModularPipelineItem = (modularPipeline) => ({
+  ...modularPipeline,
+  type: 'modularPipeline',
+  visibleIcon: VisibleIcon,
+  invisibleIcon: InvisibleIcon,
+  active: false,
+  selected: false,
+  faded: false,
+  visible: true,
+  disabled: false,
+  unset: false,
+  checked: true,
+});
+
+/**
+ * returns the corresponding parent modular pipelines
+ * for all filtered modular pipelines
+ */
+export const getFilteredModularPipelineParent = createSelector(
+  [
+    getFilteredModularPipelineItems,
+    getFilteredNodeModularPipelines,
+    (state) => state.modularPipelines,
+  ],
+  (
+    filteredModularPipelines,
+    filteredNodeModularPipelines,
+    modularPipelines
+  ) => {
+    const filteredMoudularPipelineParents = [];
+    const filteredModularPipeline = filteredModularPipelines.modularPipeline;
+
+    // 1. extract only modular pipelines with additional namespace
+    const childrenModularPipelines = filteredModularPipeline.filter(
+      (modularPipeline) => modularPipeline.id.includes('.')
+    );
+
+    // extract the parents only for those modular pipelines that does not have a filtered parent
+    childrenModularPipelines.forEach((childrenModularPipeline) => {
+      const levels = childrenModularPipeline.id.match(/\./g)
+        ? childrenModularPipeline.id.match(/\./g).length
+        : 0;
+
+      let lastIndex = 0;
+
+      for (let i = 0; i <= levels - 1; i++) {
+        // obtain the name of that pipeline
+        let parent = childrenModularPipeline.id.substr(
+          0,
+          childrenModularPipeline.id.indexOf('.', lastIndex)
+        );
+
+        // check against the filtered modular pipeline, existing list of parent pipelines and also the filtered node parent list
+        if (
+          !filteredModularPipeline.some(
+            (modularPipeline) => modularPipeline.id === parent
+          ) &&
+          !filteredMoudularPipelineParents.some(
+            (modularPipeline) => modularPipeline.id === parent
+          ) &&
+          !filteredNodeModularPipelines.some(
+            (modularPipeline) => modularPipeline.id === parent
+          )
+        ) {
+          // add the relevant modular pipeline to the list of parents
+          filteredMoudularPipelineParents.push(
+            // construct the item needed and then add it to the list
+            constructModularPipelineItem(
+              modularPipelines.filter(
+                (rawModularPipeline) => rawModularPipeline.id === parent
+              )[0]
+            )
+          );
+        }
+        lastIndex = childrenModularPipeline.id.indexOf('.', lastIndex) + 1;
+      }
+    });
+
+    return filteredMoudularPipelineParents;
+  }
+);
+
+/**
+ * returns the corresponding final set of modular pipelines
```

Comment:
Where practical please try use JSDoc format (as used elsewhere) - I know this can be tricky and some places have it missing, but still worth trying to keep it up especially for important functions like this, as well as maintaining any existing docs when changing things.